### ISSUE TITLE
Capture sample/occurrence metadata

### ISSUE THREAD

**Issue by kazlauskis** [Indicia-Team/warehouse#135]
Hi, I just wonder if there is a way to capture/store more information about the record without displaying that to users (through reports or other exports)? i.e. sample/occurrence having a metadata attribute that one could store some private-system information.

This is mainly about app records. I would like to capture more information on how the record was taken: create/update/submit times, how many submit tries, device model and version, time to geolocate, what is the photo source etc. For some of the stuff I am currently using Google Analytics, it helps to debug and understand how people use the apps, but I feel it could better be stored in our own db.

Thanks!

**Comment by johnvanbreda** [comment]
If we add a metadata custom attribute, then it could easily be hidden from things like the record details page, but it would be part of downloads (since these can be configured to automatically include all custom attributes).
An alternative would be an occurrences.metadata field added to the core data model. It could perhaps be JSON.

**Comment by kazlauskis** [comment]
Yes, I wouldn't like that system data to be downloadable or exported to other systems, so perhaps your second suggestion with a new field in the core would be better. 

There is no rush for this functionality, but I would just find it useful if it was there.

**Comment by JimBacon** [comment]
It doesn't feel quite right to me that some diagnostic information for the benefit of the developer should be in the core occurrences table. It might also be an overhead that we would want to be able to enable and disable.
I also feel that this is encroaching on user privacy and that users would have to be made aware that additional information is being collected, perhaps via an opt in to an improvement programme, before proceeding. Refer to https://ico.org.uk/media/for-organisations/documents/1596/privacy-in-mobile-apps-dp-guidance.pdf

**Comment by kazlauskis** [comment]
It could be in a form of a log, I agree. That way we could make it anonymous, but the the main idea is to be able to check what has gone wrong in case of a corrupted record or some error, for which I need to link the record to its log.

**Comment by JimBacon** [comment]
An additional table added by a warehouse module perhaps, if that is not over complicating things. Because this would not be anonymous data and because it is data that is not obvious to the user or fundamental to the app it just needs to be handled with care.

**Comment by kazlauskis** [comment]
Yes, you are right, thanks 👍
